google-ads-examples/src/main/java/com/google/ads/googleads/examples/optimization/AcquireOptimizations.java

```diff
@@ -0,0 +1,439 @@
+import com.google.ads.googleads.lib.GoogleAdsClient;
+import com.google.ads.googleads.lib.utils.FieldMasks;
+import com.google.ads.googleads.v10.enums.RecommendationTypeEnum.RecommendationType;
+import com.google.ads.googleads.v10.errors.GoogleAdsError;
+import com.google.ads.googleads.v10.errors.GoogleAdsException;
+import com.google.ads.googleads.v10.resources.Campaign;
+import com.google.ads.googleads.v10.resources.Customer;
+import com.google.ads.googleads.v10.resources.CustomerClient;
+import com.google.ads.googleads.v10.resources.Recommendation;
+import com.google.ads.googleads.v10.resources.Recommendation.RecommendationImpact;
+import com.google.ads.googleads.v10.resources.Recommendation.RecommendationMetrics;
+import com.google.ads.googleads.v10.resources.RecommendationName;
+import com.google.ads.googleads.v10.services.GoogleAdsRow;
+import com.google.ads.googleads.v10.services.GoogleAdsServiceClient;
+import com.google.ads.googleads.v10.services.GoogleAdsServiceClient.SearchPagedResponse;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.FieldMask;
+import de.siegmar.fastcsv.writer.CsvWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Acquires the suggested optimizations (optimization scores and recommendations) using the Google
+ * Ads Query Language (GAQL), and saves them into the recommendation reports of CSV format.
+ *
+ * <p>This is intended to be used with ApplyRecommendations.java which reads the recommendation
+ * reports and applies the loaded recommendations.
+ */
+public class AcquireOptimizations {
+
+  static final String DEFAULT_REPORT_DIRECTORY =
+      Paths.get(System.getProperty("user.home"), "opti-reports").toString();
+
+  static final String RECOMMENDATION_DIR_PREFIX = "recommendation_";
+  static final String RECO_OPTIMIZATION_SCORE_CSV = "optiScore.csv";
+  static final String[] RECO_OPTIMIZATION_SCORE_CSV_COLUMNS = {"CID", "AccountName", "OptiScore"};
+  static final String RECO_RECOMMENDATIONS_CSV = "recommendations.csv";
+  static final String[] RECO_RECOMMENDATIONS_CSV_COLUMNS = {
+    "ID",
+    "Type",
+    "Impact",
+    "Details",
+    "CampaignId",
+    "CampaignOptiScore",
+    "BaseImpressions",
+    "PotentialImpressions",
+    "BaseClicks",
+    "PotentialClicks",
+    "BaseCost",
+    "PotentialCost",
+    "BaseConversions",
+    "PotentialConversions",
+    "BaseVideoViews",
+    "PotentialVideoViews"
+  };
+
+  private static class AcquireOptimizationsParams extends CodeSampleParams {
+
+    @Parameter(
+        names = ArgumentNames.CUSTOMER_IDS,
+        description =
+            "Optionally specify a comma-separated list of customer IDs to retrieve recommendations"
```

Comment:
It'd be helpful if these descriptions indicated they need to either specify one or the other.

So perhaps:

`"Specify a comma-separated list of customer IDs to retrieve recommendations from. You must provide at least one of " + ArgumentNames.CUSTOMER_IDS + " or " + ArgumentNames.LOGIN_CUSTOMER_ID + "."`